types/codemirror/index.d.ts

```diff
@@ -894,16 +894,16 @@ declare namespace CodeMirror {
                 /** As with markText, this determines whether mouse events on the widget inserted for this bookmark are handled by CodeMirror. The default is false. */
                 handleMouseEvents?: boolean;
             },
-        ): CodeMirror.TextMarker;
+        ): CodeMirror.TextMarker<Position>;
 
         /** Returns an array of all the bookmarks and marked ranges found between the given positions. */
-        findMarks(from: CodeMirror.Position, to: CodeMirror.Position): TextMarker[];
+        findMarks(from: CodeMirror.Position, to: CodeMirror.Position): TextMarker<MarkerRange | Position>[];
 
         /** Returns an array of all the bookmarks and marked ranges present at the given position. */
-        findMarksAt(pos: CodeMirror.Position): TextMarker[];
+        findMarksAt(pos: CodeMirror.Position): TextMarker<MarkerRange | Position>[];
 
         /** Returns an array containing all marked ranges in the document. */
-        getAllMarks(): CodeMirror.TextMarker[];
+        getAllMarks(): CodeMirror.TextMarker<MarkerRange | Position>[];
```

Comment:
I'm not familiar with CodeMirror, but this API seems strange to me - it really returns an array of objects of two different types that have to be inspected individually before they can be accessed?  It seems like returning a range with an empty end position would be a lot less error-prone.

Reply:
I'm not gonna argue the strangeness! I got quite confused about this (in the dark times, before I used TypeScript) and wrote some funky code, but here's a screenshot demonstrating this:

![screenshot of pseuCo.com and dev tools](https://user-images.githubusercontent.com/3078885/106736336-9da33f80-6615-11eb-8c1b-7b10482c65d1.png)

What you're seeing here is a CodeMirror instance, housed in the green “pseuCo” window in a custom web application. It shows two bookmarks, the red cursor-like insert (showing the location of a syntax error) and the pencil (a position that can be selected by pressing tab, to jump behind the inserted snippet). There's also a range, namely `/* statements */`, a placeholder that can be selected with tab, too. (`/* condition */` is merely selected.)
The dev tools show the output of `cm.getAllMarks().map((m) => m.find())` (with `cm` being grabbed out of the React dev tools before). Indeed, there are two `Position`s and one `MarkerRange`.

Just for completeness, the web app in question has its source code [here](https://dgit.cs.uni-saarland.de/pseuco/concurrent-programming-web), in the `react` branch, with the CodeMirror-hosting component (in a submodule) being defined [here](https://dgit.cs.uni-saarland.de/pseuco/pseuco-shared-components/-/blob/994cfe5dd81321f386cc8c88bd22f7935b2e0dcd/ui/editors/text/TextEditor.tsx). It's deployed [here](https://react.pseuco.com/).

Reply:
Weird.  I appreciate the effort you put into confirming it - thanks!